tools/topology/topology2/platform/intel/dmic-generic.conf

```diff
@@ -76,10 +76,9 @@ IncludeByKey.PASSTHROUGH {
 			{
 				format		$FORMAT
 				index 		$DMIC0_HOST_PIPELINE_ID
-
+				core_id		$DMIC_CORE_ID
 				Object.Widget.host-copier.1 {
 					stream_name $DMIC0_PCM_CAPS
-					core_id $DMIC_CORE_ID
```

Comment:
@ranj063 for my education, how does the pipeline-level core_id get mirrored in each module instance? is this something done by the topology parser or something done by the driver+firmware?

Reply:
@plbossart it done by the topology compiler. All children objects automatically inherit their parent object's attribute values unless they have them set explicitly too.

Reply:
what happens if the base class for an object has a different value to that of the object parent? Who wins? The parent or the base class?

Reply:
Base class

Reply:
The topology2 documentation has the information on the order of priority for where the value is inferred from.

Reply:
I am sure everyone has read that documentation, right? I wouldn't even know where to find a published version.

Reply:
@plbossart there's preliminary documentation for topology2but we're going to enhance it to include all recently introduced features. Please stay tuned.